### PR TITLE
Update docs for release and dev Helm charts

### PR DESCRIPTION
**Pull request by bfjelds**
**What this PR does / why we need it**:
Introduce new Helm akri/akri-dev charts in user-guide and change our documentation to use akri-dev until our next release creates the first akri chart.

**If applicable**:
- [ ] this PR contains documentation
- [ ] this PR contains unit tests
- [ ] added code adheres to standard Rust formatting (`cargo fmt`)
- [ ] code builds properly (`cargo build`)
- [ ] code is free of common mistakes (`cargo clippy`)
- [ ] all Akri tests succeed (`cargo test`)
- [ ] inline documentation builds (`cargo doc`)
- [ ] version has been updated appropriately (`./version.sh`)

### CODE REVIEW

File: docs/user-guide.md [hunk 1]
```diff
@@ -4,6 +4,21 @@ To best understand the benefits of Akri and jump into using it, we recommend you
 ## Getting Started
 To get started using Akri, you must first decide what you want to discover and whether Akri current supports a protocol that can be used to discover resources of that type. To see the list of currently supported protocols, see our [roadmap](./roadmap.md).
 
+### Understanding Akri Helm charts
+Akri is most easily deployed with Helm charts.  Helm charts provide convenient packaging and configuration.
+
+Starting in v0.0.36, there will be a Helm chart published in **akri-dev** for each build version.  Each Akri build is verifeid with end-to-end tests on Kuberenetes, K3s, and MicroK8s, but these are likely to be less stable than our Releases.  You can deploy these versions of Akri with this command (note: **akri-dev**):
```
Comment (bfjelds) [inline docs/user-guide.md:10]: typo: verified

Comment (bfjelds) [inline docs/user-guide.md:10]: typo: Kubernetes

File: docs/user-guide.md [hunk 2]
```diff
@@ -4,6 +4,21 @@ To best understand the benefits of Akri and jump into using it, we recommend you
...
+```sh
+helm repo add akri-helm-charts https://deislabs.github.io/akri/
+helm install akri akri-helm-charts/akri-dev
+```
+
+After v0.0.36, a Helm chart will be published in **akri** for each release.  Releases will generally reflect milestones and will have more rigorous testing.  You can deploy release versions of Akri with this command (note: **akri**):
```
Comment (bfjelds) [inline docs/user-guide.md:16]: pick a case for 'release' (or 'Release')

File: docs/user-guide.md (hunk 1 above)
Comment (kate-goldenring) [inline docs/user-guide.md:10]: "verified" and "Kubernetes".
Also, this is a great [VSCode extension](https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker) for catching spelling errors!

File: docs/user-guide.md (hunk 2 above)
Comment (jiria) [inline docs/user-guide.md]: Should this say "in akri <what> for each release" ?

**Review by jiria — APPROVED** [review]

File: docs/user-guide.md (hunk 1 above)
Comment (bfjelds) [reply docs/user-guide.md]: nice!

File: docs/user-guide.md (hunk 2 above)
Comment (bfjelds) [reply docs/user-guide.md]: do you mean without the emphasis?

Comment (jiria) [reply docs/user-guide.md]: Sorry, I am not very good with the clarity these days... Also, there was an extra token which got deleted by the rendering. 

`in **akri** <what> for each release `

where `<what>` should be repo/chart/something. Or is it just me that it feels unqualified?

Comment (bfjelds) [reply docs/user-guide.md]: I had stream and feed in a few drafts, but figured nothing was just as good.

Happy to change to whatever feels better or to the right word if someone knows it.

Comment (jiria) [reply docs/user-guide.md]: Would this make sense?

After v0.0.36, **akri** Helm chart will be published for each release.

**Pull request merged**